### PR TITLE
Various fixes to contrib/unbound_munin_ file

### PR DESCRIPTION
**Pull request by kimheino**
The most important fix is "fix statistics after unbound restart / server reboot". Other changes are mostly cosmetic.

### CODE REVIEW

File: contrib/unbound_munin_ [hunk 1]
```diff
@@ -98,7 +95,9 @@ BSD
 
 =cut
 
-state=${statefile:-/usr/local/var/munin/plugin-state/unbound-state}
+. ${MUNIN_LIBDIR}/plugins/plugin.sh
```
Comment (gthess) [inline contrib/unbound_munin_]: Why do we need plugin.sh? I don't think we use anything from there, right?

File: contrib/unbound_munin_ [hunk 2]
```diff
@@ -98,7 +95,9 @@ BSD
...
+state="${MUNIN_PLUGSTATE}/unbound.state"
+seentags="${MUNIN_PLUGSTATE}/unbound-seentags.state"
```
Comment (gthess) [inline contrib/unbound_munin_]: Are these environment variables guaranteed to be set?

File: contrib/unbound_munin_ [hunk 3]
```diff
@@ -467,27 +475,23 @@ memory)
...
+	for nm in `grep "^num.query.opcode" $seentags`; do
+		print_value $nm
 	done
 	;;
 by_rcode)
-	for x in `grep "^num.answer.rcode" $state`; do
-		nm=`echo $x | sed -e 's/=.*$//'`
-		print_value_line $nm $x
+	for nm in `grep "^num.answer.rcode" $seentags`; do
+		print_value $nm $x
```
Comment (gthess) [inline contrib/unbound_munin_]: stray $x?

File: contrib/unbound_munin_ [hunk 4]
```diff
@@ -121,12 +120,24 @@ get_value ( ) {
...
+update_seentags() {
+    tmplist="$(cat ${seentags} 2> /dev/null)
+num.query.type.A
+num.query.class.IN
+num.query.opcode.QUERY
+num.answer.rcode.NOERROR
+"
+    (echo "${tmplist}"; grep ^num ${state} | sed -e 's/=.*//') | sort -u > ${seentags}
+}
+
```
Comment (gthess) [inline contrib/unbound_munin_:133]: OK, so here is the core of the change. If I understand correctly you want munin to keep plotting tags that unbound was reporting on before the restart (with 0 as the value if the tag is no longer reported by unbound; based on get_value()).

But why do you always set the above 4?

**Review by gthess — COMMENTED** [review]
Thanks for taking the time for this!
I have some questions inline before accepting this.

Not a munin user but I am worried if this change catches other users by surprise or if it is not relevant to other users.
Do you see value in having an environment variable to toggle this new behavior? Then based on the environment variable update_seentags() could just write the state content (filtered) to the seentags file and the old behavior is also kept by default.

**Comment by kimheino** [comment]
> Not a munin user but I am worried if this change catches other users by surprise or if it is not relevant to other users.
> Do you see value in having an environment variable to toggle this new behavior? Then based on the environment variable update_seentags() could just write the state content (filtered) to the seentags file and the old behavior is also kept by default.

Old behavior has two fatal errors:

1) Munin reports error if you restart unbound (tag vanishes). 
2) New tags will not be included to Munin's graph if they are not present at the first run of munin-async. So basically graph stays empty if you reboot and are using munin-async.

New behavior might include some tags that stays zero forever, but that should not matter. So no, I don't see any point including config option for switching between old/new.

I'll reply inline for others.

File: contrib/unbound_munin_ (hunk 1 above)
Comment (kimheino) [reply contrib/unbound_munin_]: You're correct, I'll remove it.

File: contrib/unbound_munin_ (hunk 2 above)
Comment (kimheino) [reply contrib/unbound_munin_]: Yes, they are always present.

File: contrib/unbound_munin_ (hunk 3 above)
Comment (kimheino) [reply contrib/unbound_munin_]: Oops, sorry, I'll remove it.

File: contrib/unbound_munin_ (hunk 4 above)
Comment (kimheino) [reply contrib/unbound_munin_:133]: Yes, you understood it correctly.

Those 4 tags are needed, as later on code has "`for nm in grep "^num.query.type" $seentags; do`" (and num.query,class, opcode and rcode). This makes it sure that there are at least one tag for each graph, needed by Munin. Graph without tag is not allowed.

**Comment by gthess** [comment]
LGTM now, thanks for explaining the reason behind the changes.

**Pull request merged**